composer/rest/composer2/composer2_airflow_rest_api.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+Trigger a DAG in Cloud Composer 2 environment using the Airflow 2 stable REST API.
+"""
+
+# [START composer_2_trigger_dag]
+
+import google.auth
+from google.auth.transport.requests import AuthorizedSession
+
+AUTH_SCOPE = "https://www.googleapis.com/auth/cloud-platform"
+
+
+def make_composer2_web_server_request(url, method="GET", **kwargs):
+    """
+    Make a request to Cloud Composer 2 environment's web server.
+    Args:
+      url: The URL to fetch.
+      method: The request method to use ('GET', 'OPTIONS', 'HEAD', 'POST', 'PUT',
+        'PATCH', 'DELETE')
+      **kwargs: Any of the parameters defined for the request function:
+                https://github.com/requests/requests/blob/master/requests/api.py
+                  If no timeout is provided, it is set to 90 by default.
+    """
+
+    credentials, _ = google.auth.default(scopes=[AUTH_SCOPE])
```

Comment:
We might want to encourage folks to use these credentials more than once, either by putting this as a global or adding a comment encouraging them to construct this at app start-up time. On some environments, it can take 30 seconds+ to get credentials.

Reply:
@leahecole Do we have an existing way to working with credentials in Composer samples? I've seen that we use comments to outline things to do for developers. Should I do likewise here?

Reply:
@pavel-salnikov not yet - we don't have samples with the client library itself yet because the library is so new, so I'm actually pretty inexperienced with this. I think I'd prefer to make this a global AND document it just to cover our bases. @tswast I'll poke through BQ samples now to see if there's examples but if you have any you like please send along!

Reply:
In my quick perusal I didn't find any good examples in samples but did find that doc I linked to that mentions this. Happy to adjust phrasing on the comment if you'd like